packages/toolpad-app/src/toolpad/AppEditor/PageEditor/RenderPanel.tsx

```diff
@@ -1318,43 +1326,59 @@ export default function RenderPanel({ className }: RenderPanelProps) {
       const cursorPos = canvasHostRef.current?.getViewCoordinates(event.clientX, event.clientY);
 
       if (draggedNodeRect && parentRect && resizePreviewElement && cursorPos) {
-        let snappedToGridCursorPosX =
-          Math.round(cursorPos.x / RESIZE_SNAP_UNITS) * RESIZE_SNAP_UNITS;
+        if (draggedEdge === RECTANGLE_EDGE_LEFT || draggedEdge === RECTANGLE_EDGE_RIGHT) {
```

Comment:
It seems that there is a lot of this "resizing" logic going on in this component. Could we maybe abstract it away to some `ResizeablePanel` component? 🤔

Reply:
in addition to the above, i haven't considered this approach so will see if there's a way to move this logic into a separate React component. thanks for the suggestion!